### ISSUE TITLE
Wrong certificate selecttion for kRSA ciphers when both rsa-pss and rsa certificates configured

### ISSUE THREAD

**Issue by tomato42** [openssl/openssl#7059]
Version: openssl-1.1.1-0.pre9.1.fc30.x86_64

Generate both regular rsaEncryption and rsassa-pss certificates, configure s_server with both of them, e.g.:
```
openssl req -x509 -newkey rsa-pss -pkeyopt rsa_keygen_bits:2048 -keyout localhost-pss.key -out localhost-pss.crt -subj /CN=localhost -nodes -batch
openssl req -x509 -newkey rsa -pkeyopt rsa_keygen_bits:2048 -keyout localhost.key -out localhost.crt -subj /CN=localhost -nodes -batch
openssl s_server -key localhost-pss.key -cert localhost-pss.crt -dkey localhost.key -dcert localhost.crt
```
Connect to it while advertising TLS 1.2, kRSA ciphers _but_ also rsa_pss_pss signature algorithms:
```
openssl s_client -cipher kRSA -tls1_2 -connect localhost:4433
```

Connection is aborted with a `missing rsa encrypting cert` error as the provided certificate is the rsa-pss one:
```
CONNECTED(00000005)
depth=0 CN = localhost
verify error:num=18:self signed certificate
verify return:1
depth=0 CN = localhost
verify return:1
140691431543808:error:140820A9:SSL routines:ssl3_check_cert_and_algorithm:missing rsa encrypting cert:ssl/statem/statem_clnt.c:3585:
---
Certificate chain
 0 s:CN = localhost
   i:CN = localhost
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIDaTCCAiCgAwIBAgIUb9y20CSpiYzYmprz6iSvDA5bcE0wPgYJKoZIhvcNAQEK
MDGgDTALBglghkgBZQMEAgGhGjAYBgkqhkiG9w0BAQgwCwYJYIZIAWUDBAIBogQC
AgDeMBQxEjAQBgNVBAMMCWxvY2FsaG9zdDAeFw0xODA4MTIxNTMzMTZaFw0xODA5
MTExNTMzMTZaMBQxEjAQBgNVBAMMCWxvY2FsaG9zdDCCASAwCwYJKoZIhvcNAQEK
A4IBDwAwggEKAoIBAQDLTcDMS7WohE+ZuU/PbEETIpInv5M+N0hsr1fDiwtA6fp+
wXBPcUJxuIlv4gULxR37a5DiJVcJTzIBigfMSkEnSAA5OutGFb0wuWMWii46CxnD
WlaNw9A0Xw5aig+Y25Fj/K7JbGG3CK1sG+KPyRG58u/d4VCdx/DNBUN37c3m3Yq2
ZlqDK7fzd/UbtAqShoVNu0GXAOh0No232MjgzzuE7HrLTgfVg/nNPrS/hvWEFVo7
FMnvAjha9cjuFJ6EWOaCfkP4WPRCREi2BHU5AIeHa5NDM3tU/c/7rKHAwIWIr6Zw
izQiGYis+W2WWf81zc0Mm9YFXiiVR1dUgV6++uOfAgMBAAGjUzBRMB0GA1UdDgQW
BBQwCZoXsEj+zBUWmdbUu/K3az5vdzAfBgNVHSMEGDAWgBQwCZoXsEj+zBUWmdbU
u/K3az5vdzAPBgNVHRMBAf8EBTADAQH/MD4GCSqGSIb3DQEBCjAxoA0wCwYJYIZI
AWUDBAIBoRowGAYJKoZIhvcNAQEIMAsGCWCGSAFlAwQCAaIEAgIA3gOCAQEAX0Up
sKTEcvXQExH4oW6JHILkay26xSNbQtOoawYn6XuC+YyW/x/eNldCOfaFNvznN3z+
5dDhAArvJKkaKRlcIFfBhIPN00E+df6Q8NwLezNGtSJp4TbWaGWUn3UnlpIZOwXh
8b7Dh0yUgWRZUmoDHjMsYAkamUpAlQ9qgI4zTdTq6q4HbfvDONmWksmDPcWHhMSW
fKNtdVDXcU33z0T4l1Lj+Db/npZ3uj6kkjIaJslWcfsWhUzRzOzPkZ5ZqjTY7iBW
W7J4NBdxoKv6F4GBZZI68RF90NM2bNDFCcreKz4gVG19rL57FHefgXkTcqSWngPx
lyckpgtLiyl6623nJg==
-----END CERTIFICATE-----
subject=CN = localhost

issuer=CN = localhost

---
No client certificate CA names sent
---
SSL handshake has read 963 bytes and written 201 bytes
Verification error: self signed certificate
---
New, (NONE), Cipher is (NONE)
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1535300836
    Timeout   : 7200 (sec)
    Verify return code: 18 (self signed certificate)
    Extended master secret: yes
---
```

**Comment by tomato42** [comment]
@openssl that feels like a 1.1.1 blocker to me, can anyone confirm?

**Comment by t-j-h** [comment]
@tomato42 could you check this behaviour in earlier releases - that would be useful to know

**Comment by t-j-h** [comment]
No particular version - just the first thing will be to track down if this is a behaviour change or not in this current pending release.

**Comment by mattcaswell** [comment]
Fix for this in #7099.

**Comment by tomato42** [comment]
@t-j-h it looks like the issue was present since introduction of support for RSA-PSS in 800c4883d0a0e98
